src/settings/toml/deploy_target.rs

```diff
@@ -61,7 +71,17 @@ impl DeployTarget {
         route_config: &RouteConfig,
     ) -> Result<DeployTarget, failure::Error> {
         if route_config.is_zoneless() {
-            Ok(DeployTarget::Zoneless)
+            // account_id is required
+            let account_id = route_config.account_id.as_ref().unwrap();
```

Comment:
can we just do `let account_id = &route_config.account_id`? if not we should do `as_ref()?` if possible

Reply:
the issue is that the type of `route_config.account_id` here is `&Option<String>`, not String. so it needs to be unwrapped; but i can't have just a naked string, so it's gotta  be an `as_ref` call, and you can't call `?` on the result of `as_ref`.

i have, however, been considering just making two constructors: one for a zoned and one for a zoneless, because by the time i call this i should know what it's gonna be.